components-compose/src/main/java/uk/gov/hmrc/components/compose/molecule/input/PasswordInputView.kt

```diff
@@ -0,0 +1,376 @@
+
+    @Composable
+    fun ShowHideButton(
+        modifier: Modifier,
+        hideButtonContentDescription: String?,
+        hideButtonStateDescription: String?,
+        showButtonContentDescription: String?,
+        showButtonStateDescription: String?,
+        showButtonText: String,
+        hideButtonText: String
+    ) {
+        if (hideButtonShown) {
+            SecondaryButton(
+                hideButtonText, onClick = { hideButtonShown = false },
+                modifier = modifier
+                    .semantics {
+                        hideButtonContentDescription?.let { contentDescription = it }
+                        hideButtonStateDescription?.let { stateDescription = it }
+                    }
+            )
+        } else {
+            SecondaryButton(
+                showButtonText, onClick = { hideButtonShown = true },
+                modifier = modifier
+                    .semantics {
+                        showButtonContentDescription?.let { contentDescription = it }
+                        showButtonStateDescription?.let { stateDescription = it }
+                    }
+            )
+        }
+    }
+
+    CalculateBorder(
+        true,
+        isError,
+        interactionSource,
+        focusedBorderThickness = FocusedBorderThickness,
+        unfocusedBorderThickness = UnfocusedBorderThickness
+    )
+    Column(modifier = modifier) {
+        Label(labelText = labelText, labelContentDescription = labelContentDescription)
+        Hint(hintText = hintText, hintContentDescription = hintContentDescription)
+        Row(
+            borderStroke?.let {
+                Modifier
+                    .fillMaxWidth()
+                    .border(it.value, RectangleShape)
+            } ?: Modifier,
+            verticalAlignment = Alignment.CenterVertically
+        ) {
+            Column(Modifier.weight(1f)) {
+                TextField(
+                    modifier = if (isError) Modifier.clearAndSetSemantics {
+                        localError?.let {
+                            error(
+                                "$it. $errorContentDescription"
+                            )
+                        }
+                    } else Modifier,
+                    isError = isError,
+                    value = localValue,
+                    onInputValueChange = { newValue ->
+                        if (maxChars?.let { newValue.length <= it } != false) {
+                            localValue = if (inputFilter != null && newValue.isNotEmpty()) {
+                                inputFilter(newValue, localValue)
+                            } else newValue
+                            if (onInputValueChange != null) {
+                                onInputValueChange(localValue)
+                            }
+                        }
+                    },
+                    prefix = prefix,
+                    placeholderText = { placeholderText?.let { Text(text = it) } },
```

Comment:
I think we can remove this because we don't want placeholder text on this component.